### DISCUSSION TITLE
How to update to Nextcloud 30 now?

### DISCUSSION THREAD

**Discussion by jakobroehrl** [nextcloud/all-in-one#5133] [Wiki]
If you don't want to wait for AIO getting updated to NC30 for everyone in the near future, you can upgrade to Nextcloud 30 already now by following these steps:

1. First, open the aio interface and create a backup and start your containers again after the backup is successful.

2. Afterwards, run via CLI:
```
# Go into the container
sudo docker exec --user www-data -it nextcloud-aio-nextcloud bash
```

3. Now inside the container:

```
# Switch to the beta channel
php occ config:system:set updater.release.channel --value=beta

# Run the update
php updater/updater.phar --no-interaction --no-backup && php occ app:enable nextcloud-aio --force

# Switch back to the stable channel
php occ config:system:set updater.release.channel --value=stable

# Exit the container
exit
```

The above instructions should upgrade you to Nextcloud 30. You might need to follow the steps a second time to reach NC30. If something goes wrong, feel free to restore the backup.

**Comment by linux-universe** [comment]
Has anyone tested this, if so were there any problems?

**Reply by ArtemStepanov** [reply]
Just tested it, for some reasons can't see the 30 version. Upgrade is still running tho.

```bash
$ php updater/updater.phar --no-interaction --no-backup && php occ app:enable nextcloud-aio --force
Nextcloud Updater - version: v29.0.0-1-g9293bcb dirty

Current version is 29.0.6.

Update to Nextcloud 29.0.7 available. (channel: "beta")
Following file will be downloaded automatically: https://download.nextcloud.com/server/releases/nextcloud-29.0.7.zip
Open changelog ↗

Updater run in non-interactive mode.

Start update

Info: Pressing Ctrl-C will finish the currently running step and then stops the updater.

[✔] Check for expected files
[✔] Check for write permissions
[✔] Create backup
[ ] Downloading ...
```

UPD: Changed the channel to Beta from NC WebUI and now it shows 30 version available to setup.
<img width="918" alt="image" src="https://github.com/user-attachments/assets/f62a9d6c-e2a7-45da-af96-959aa13b149d">

**Reply by ghost** [reply]
I just did it a few minutess ago. You have to run it a second time afterwards. There is a little note about it at the end of the insturction. Also mist that on my first update step.

**Reply by ArtemStepanov** [reply]
Just updated it, do not recommend doing that at the moment, since most of apps have no support of 30 version.

**Comment by ghost** [comment]
I've updated to NC30 using the steps provided and all went fine.
However the webinterface will now say:
```
Detected some missing optional indices. Occasionally new indices are added (by Nextcloud or installed applications) to improve database performance. Adding indices can sometimes take awhile and temporarily hurt performance so this is not done automatically during upgrades. Once the indices are added, queries to those tables should be faster. Use the command `occ db:add-missing-indices` to add them. Missing indices: "mail_messages_mb_id_uid_uidx" in table "mail_messages", "mail_smime_certs_uid_email_idx" in table "mail_smime_certificates", "mail_trusted_senders_idx" in table "mail_trusted_senders", "mail_coll_idx" in table "mail_coll_addresses".
```
And attempting the command `php occ db:add-missing-indices` does not resolve it.
So far I haven't been able to get rid of this message.

Also you'll need to create an empty `.ocdata` file inside the root of your nextcloud data directory.
This file is now required for the borgbackup thing to work.

Asside from the missing indices warning, it seems to work fine.

**Comment by owerrride** [comment]
Its works, great

**Comment by gbcatrinoiu** [comment]
Is the official update going to come out soon?

**Reply by linux-universe** [reply]
https://github.com/nextcloud/all-in-one/issues/5249
This is the to-do list

**Comment by daks** [comment]
I tested this procedure but it upgraded to 29.0.8 not 30

```
Current version is 29.0.7.

Update to Nextcloud 29.0.8 available. (channel: "beta")
Following file will be downloaded automatically: https://download.nextcloud.com/server/releases/nextcloud-29.0.8.zip
```

**edit**: after running the `php updater/updater.phar --no-interaction --no-backup && php occ app:enable nextcloud-aio --force` several times the version is now 30.0.1 RC2

**Reply by srvprivate** [reply]
Latest is working as of last night/this morning 30.0.1

**Reply by g-work** [reply]
Tried 30.0.2 today. Works.

**Comment by tekh** [comment]
After increasing the PHP_MEMORY_LIMIT I was able to upgrade to 29.0.8 and 30.0.2 afterwards. The update was failing after the download; [ ] Verify integrity ...PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 239166008 bytes) in phar:///var/www/html/updater/updater.phar/lib/Updater.php on line 674

sudo -u www-data PHP_MEMORY_LIMIT=1024M php updater/updater.phar --no-interaction --no-backup && sudo -u www-data PHP_MEMORY_LIMIT=1024M php occ app:enable nextcloud-aio --force

**Reply by szaimen** [reply]
Hi, you are running the occ command incorrectly. If you want to use sudo you need to use the `-E` option to keep environmental varialbles to circumvent your issue.

**Reply by tekh** [reply]
I learned something today. Thank you

**Comment by romainmp** [comment]
My AIO containers apparently cannot resolve DNS:
`In Updater.php line 509:`
`  Could not do request to updater server: Could not resolve host: updates.nextcloud.com`

I'm trying to install on Truenas Scale 24.10 ; any idea ?

**Reply by romainmp** [reply]
Nevermind, it was a config missing for the containers to resolve DNS

**Reply by Techttv** [reply]
How did you solve? Same error

**Comment by szaimen** [comment]
Nc30 is now released with v10.0.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
